src/AutoRest.CSharp.V3/Generation/Writers/RestClientWriter.cs

```diff
@@ -191,6 +191,43 @@ private void WriteRequestCreation(CodeWriter writer, RestClientMethod clientMeth
                             writer.Line($");");
                         }
                         break;
+                    case MultipartRequestBody multipartRequestBody:
+                        var multipartContent = new CodeWriterDeclaration("content");
+                        writer.Line($"var {multipartContent:D} = new {typeof(MultipartFormDataContent)}();");
+
+                        foreach (var bodyParameter in multipartRequestBody.Value)
+                        {
+                            using (WriteValueNullCheck(writer, bodyParameter))
+                            {
+                                if (bodyParameter.Type.Name == "IEnumerable")
+                                {
+                                    var collectionItemVariable = new CodeWriterDeclaration("value");
+                                    using (writer.Scope($"foreach (var {collectionItemVariable:D} in {bodyParameter.Reference.Name})"))
+                                    {
+                                        writer.Append($"{multipartContent}.Add({typeof(RequestContent)}.Create({collectionItemVariable}), \"");
+                                        WriteConstantOrParameter(writer, bodyParameter);
+                                        writer.Line($"\", null);");
+                                    }
+                                }
+                                else
+                                {
+                                    if (bodyParameter.Type.Name == typeof(string).Name)
```

Comment:
We shouldn't hardcode this check here.

Let's make `MultipartRequestBodyPart.Content` to be of RequestBody type, then we can reuse the logic from https://github.com/Azure/autorest.csharp/pull/886/files#diff-f1877212f4818ce2519525272e8e863eR178-R193 to create it.

Reply:
This way when somebody asks for a model to be part of multipart later things would "just work".